src/main/java/org/privacyidea/JSONParser.java

```diff
@@ -0,0 +1,345 @@
+package org.privacyidea;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSyntaxException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.privacyidea.PIConstants.ASSERTIONCLIENTEXTENSIONS;
+import static org.privacyidea.PIConstants.ATTRIBUTES;
+import static org.privacyidea.PIConstants.AUTHENTICATORDATA;
+import static org.privacyidea.PIConstants.CLIENTDATA;
+import static org.privacyidea.PIConstants.CODE;
+import static org.privacyidea.PIConstants.CREDENTIALID;
+import static org.privacyidea.PIConstants.DETAIL;
+import static org.privacyidea.PIConstants.ERROR;
+import static org.privacyidea.PIConstants.ID;
+import static org.privacyidea.PIConstants.INFO;
+import static org.privacyidea.PIConstants.JSONRPC;
+import static org.privacyidea.PIConstants.MAXFAIL;
+import static org.privacyidea.PIConstants.MESSAGE;
+import static org.privacyidea.PIConstants.MESSAGES;
+import static org.privacyidea.PIConstants.MULTI_CHALLENGE;
+import static org.privacyidea.PIConstants.OTPLEN;
+import static org.privacyidea.PIConstants.REALMS;
+import static org.privacyidea.PIConstants.RESULT;
+import static org.privacyidea.PIConstants.SERIAL;
+import static org.privacyidea.PIConstants.SIGNATURE;
+import static org.privacyidea.PIConstants.SIGNATUREDATA;
+import static org.privacyidea.PIConstants.STATUS;
+import static org.privacyidea.PIConstants.TOKEN;
+import static org.privacyidea.PIConstants.TOKENS;
+import static org.privacyidea.PIConstants.TOKEN_TYPE_WEBAUTHN;
+import static org.privacyidea.PIConstants.TRANSACTION_ID;
+import static org.privacyidea.PIConstants.TYPE;
+import static org.privacyidea.PIConstants.USERHANDLE;
+import static org.privacyidea.PIConstants.USERNAME;
+import static org.privacyidea.PIConstants.VALUE;
+import static org.privacyidea.PIConstants.VERSION_NUMBER;
+import static org.privacyidea.PIConstants.WEBAUTHN_SIGN_REQUEST;
+
+public class JSONParser {
+
+    private final PrivacyIDEA privacyIDEA;
+
+    public JSONParser(PrivacyIDEA privacyIDEA) {
+        this.privacyIDEA = privacyIDEA;
+    }
+
+    public String formatJson(String json) {
+        if (json == null || json.isEmpty()) return "";
+
+        JsonObject obj;
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        try {
+            obj = JsonParser.parseString(json).getAsJsonObject();
+        } catch (JsonSyntaxException e) {
+            privacyIDEA.error(e);
+            return json;
+        }
+
+        return gson.toJson(obj);
+    }
+
+    String parseAuthToken(String serverResponse) {
```

Comment:
So this method actually parses the response of the /auth request and not the JWT itself?
Maybe I would have expected another naming.

Reply:
it extracts the auth token from the response, so maybe extractAuthToken would be a more appropriate name.

Reply:
OK, I understand.
I like "get"s ;-) - sounds like "get the auth token from the response (parameter)".
Just a comment, you choose, if you like "parse", "extract"....